server/src/test/scala/services/crunch/FlightUpdatesTriggerNewPortStateSpec.scala

```diff
@@ -19,80 +19,125 @@ class FlightUpdatesTriggerNewPortStateSpec extends CrunchTestLike {
-        terminalProcessingTimes = Map(T1 -> Map(
-          eeaMachineReadableToDesk -> 25d / 60,
-          eeaMachineReadableToEGate -> 25d / 60
-        )),
-        queuesByTerminal = SortedMap(T1 -> Seq(EeaDesk, EGate))
-      ))
-
-    offerAndWait(crunch.liveArrivalsInput, ArrivalsFeedSuccess(inputFlightsBefore))
-    offerAndWait(crunch.liveArrivalsInput, ArrivalsFeedSuccess(inputFlightsBefore))
-    offerAndWait(crunch.liveArrivalsInput, ArrivalsFeedSuccess(inputFlightsAfter))
-
-    val expectedFlights = Set(ApiFlightWithSplits(
-      updatedArrival.copy(FeedSources = Set(LiveFeedSource)),
-      Set(Splits(Set(ApiPaxTypeAndQueueCount(EeaMachineReadable, Queues.EeaDesk, 100.0, None)), TerminalAverage, None, Percentage))))
-
-    crunch.portStateTestProbe.fishForMessage(3 seconds) {
-      case ps: PortState =>
-        val flightsAfterUpdate = ps.flights.values.map(_.copy(lastUpdated = None)).toSet
-        flightsAfterUpdate == expectedFlights
+  "Given a noop update to an existing flight followed by a real update " >> {
+    "When I expect a PortState " >> {
+      "Then I should see one containing the updated flight" >> {
+
+        val scheduled = "2017-01-01T00:00Z"
+
+        val flight = ArrivalGenerator.arrival(schDt = scheduled, iata = "BA0001", terminal = T1, actPax = Option(21))
+        val inputFlightsBefore = Flights(List(flight))
+        val updatedArrival = flight.copy(ActPax = Some(50))
+        val inputFlightsAfter = Flights(List(updatedArrival))
+        val crunch = runCrunchGraph(
+          now = () => SDate(scheduled),
+          airportConfig = defaultAirportConfig.copy(
+            terminalProcessingTimes = Map(T1 -> Map(
+              eeaMachineReadableToDesk -> 25d / 60,
+              eeaMachineReadableToEGate -> 25d / 60
+              )),
+            queuesByTerminal = SortedMap(T1 -> Seq(EeaDesk, EGate))
+            ))
+
+        offerAndWait(crunch.liveArrivalsInput, ArrivalsFeedSuccess(inputFlightsBefore))
+        offerAndWait(crunch.liveArrivalsInput, ArrivalsFeedSuccess(inputFlightsBefore))
+        offerAndWait(crunch.liveArrivalsInput, ArrivalsFeedSuccess(inputFlightsAfter))
+
+        val expectedFlights = Set(ApiFlightWithSplits(
+          updatedArrival.copy(FeedSources = Set(LiveFeedSource)),
+          Set(Splits(Set(ApiPaxTypeAndQueueCount(EeaMachineReadable, Queues.EeaDesk, 100.0, None)), TerminalAverage, None, Percentage))))
+
+        crunch.portStateTestProbe.fishForMessage(3 seconds) {
+          case ps: PortState =>
+            val flightsAfterUpdate = ps.flights.values.map(_.copy(lastUpdated = None)).toSet
+            flightsAfterUpdate == expectedFlights
+        }
+
+        success
+      }
     }
+  }
 
-    crunch.shutdown
-
-    success
+  "Given an existing ACL flight and crunch data" >> {
+    "When I send an empty set of ACL flights" >> {
+      "Then I should see the pax nos and workloads fall to zero for the flight that was removed" >> {
```

Comment:
This is the new test to check the pax & workload are zero'd out when we get no flights to crunch after previously having some